File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        chars, ans = defaultdict(list), -1
        
        for i in range(len(s)):
            chars[s[i]].append(i)
            
        for key, val in chars.items():
            if len(val) == 1:
                ans = val[0] if ans == -1 else min(ans, val[0])
            
        return ans
        

File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        map<char, vector<int>> dict;
        int ans = -1;
        
        for (int i = 0; i < s.size(); ++i)
            dict[s[i]].push_back(i);
        
        for (auto& [_, val]:dict){
            if (val.size() == 1)
                ans = (ans == -1)? val[0]:min(ans,val[0]);
        }
        
        return ans;
    }
};